bitcoin/src/pow.rs

```diff
@@ -248,14 +257,36 @@ impl Target {
     ///
     /// The difficulty can only decrease or increase by a factor of 4 max on each difficulty
     /// adjustment period.
-    pub fn min_difficulty_transition_threshold(&self) -> Self { Self(self.0 >> 2) }
+    pub fn min_difficulty_transition_threshold(&self) -> Self {
+        let min = self.0 >> 2;
+
+        if min < U256::ONE {
+            Self(U256::ONE)
+        } else {
+            Self(min)
```

Comment:
FYI IIRC `max` and `min` are methods on `PartialOrd` and since `U256` implements it you should be able to write `min.max(U256::ONE)`.

Reply:
Ah thanks, this was super ugly before.